site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java

```diff
@@ -15151,6 +15156,54 @@ private Locale getLocaleFromString(String localeString)
 	}
 	
 	/**
+	 * Handle the eventSubmit_doUnjoin command to have the user un-join this site.
+	 */
+	public void doUnjoin(RunData data) {
+		
+		SessionState state = ((JetspeedRunData) data)
+			.getPortletSessionState(((JetspeedRunData) data).getJs_peid());
+		final ParameterParser params = data.getParameters();
+
+		final String id = params.get("itemReference");
+		String siteTitle = null;
+		
+		if (id != null)	{
+			try	{
+				siteTitle = SiteService.getSite(id).getTitle();
+                System.out.println("siteTitle:" + siteTitle);
+				SiteService.unjoin(id);
+				String msg = rb.getString("sitinfimp.youhave") + " " + siteTitle;
+				addAlert(state, msg);
+				
+			} catch (IdUnusedException ignore) {
+				
+			} catch (PermissionException e)	{
+				// This could occur if the user's role is the maintain role for the site, and we don't let the user
+				// unjoin sites they are maintainers of
+				Log.warn("chef", this + ".doUnjoin(): " + e);
+				//TODO can't access site so redirect to portal
+				
+			} catch (InUseException e) {
+				addAlert(state, siteTitle + " "
+						+ rb.getString("sitinfimp.sitebeing") + " ");
+			}
+		}
+		
+		// refresh the whole page
+		scheduleTopRefresh();
+		
+		/*
+		 * It would have been nice to redirect to the portal 
+		 * if the user no longer has access to view the site;
+		 * but it seems that it is not possible in velocity
+		 * to redirect if action is set.
+		 * 
+		*/
```

Comment:
They get redirected to the same site and get the join dialog - I'll just strip that comment out. It dates back to 2.5.